feat: Implement a method to change the device id

DeboraTahara opened the pull request:
<!--
Depends on:
- [ ]

-->
# What is this PR for
[sc-65148]

# How I did it
<!-- Explain a little bit of your implementation -->

# Results, How I tested
I already tested it on Tupan 3.

# Checklist
- [x] Assign yourself  in the PR

DeboraTahara committed:
feat: Implement a method to change the device id

doudou commented on src/Main.cpp:
```diff
@@ -96,6 +96,18 @@ int main(int argc, char** argv)
         }
         std::cout << wp_device.getConfig() << std::endl;
     }
+    else if (cmd == "set_id") {
```
Also update the usage message at the top of the file please (add the `dc-cube` command as well)

doudou replied:
Ping @DeboraTahara

wvmcastro committed:
fix: write set id query

wvmcastro commented on src/Main.cpp:
```diff
@@ -96,6 +96,20 @@ int main(int argc, char** argv)
+        waitResult(*can_device, device, device.querySerialNumber());
+        uint32_t serial_number = device.getSerialNumber();
+        int device_new_id(std::stoi(argv[6]));
+        waitResult(*can_device, device, device.querySetId(device_new_id, serial_number));
```
we verified in live that we should send the set id query to can id 0x600 (broadcast)

DeboraTahara committed:
fix: Hard coded the can id broadcast

DeboraTahara committed:
feat: Add commands dc-cube and set-id to command list

doudou commented on src/Main.cpp:
```diff
@@ -96,6 +98,20 @@ int main(int argc, char** argv)
+        waitResult(*can_device, device, device.querySerialNumber());
+        uint32_t serial_number = device.getSerialNumber();
+        int device_new_id(std::stoi(argv[6]));
+        waitResult(*can_device, device, device.querySetId(0x600, serial_number));
```
This makes no sense ... the new ID is not given anywhere in the package.

DeboraTahara replied:
I agree with you, but it what I understood about 
> we verified in live that we should send the set id query to can id 0x600 (broadcast)
@wvmcastro can you explain how did you test it?

wvmcastro commented on src/Main.cpp:
```diff
@@ -96,6 +98,20 @@ int main(int argc, char** argv)
         std::cout << wp_device.getConfig() << std::endl;
     }
+    else if (cmd == "set-id") {
+        if (argc != 7)
+        {
+            std::cerr << "wrong number of arguments" << std::endl;
+        }
+        power_whisperpower::Device device(device_group, device_id);
+        waitResult(*can_device, device, device.querySerialNumber());
+        uint32_t serial_number = device.getSerialNumber();
+        int device_new_id(std::stoi(argv[6]));
+        waitResult(*can_device, device, device.querySetId(0x600, serial_number));
```
```suggestion
        power_whisperpower::Device wp_device(device_group, 0x600);
        waitResult(*can_device, wp_device, wp_device.querySerialNumber());
        uint32_t serial_number = device.getSerialNumber();
        int device_new_id(std::stoi(argv[5]));
        waitResult(*can_device, wp_device, device.querySetId(device_new_id, serial_number));
```

also write a disclaimer at the helpers stating that when using set-id; the `DEVICE_ID` argument is actually the _NEW DEVICE ID_ to be set

doudou committed:
chore: update CMake rock_init call

doudou committed:
chore: list the name of the device types in power_whisperpower_ctl

doudou committed:
chore: report proper error if the CAN device cannot be found

Instead of SEGV

doudou committed:
fix: querySetId is not device-bound, make it a static method

We have to send it to can_id=0x600, and the serial number seems to be completely
unused. Just hard-code the message that has to be sent, and make sure the caller
is aware that it should really be done only if there is one device on the bus.

doudou commented:
@DeboraTahara I updated it to the (we believe) working version.

@wvmcastro @higor-melo please review

wvmcastro commented on src/Device.cpp:
```diff
@@ -42,6 +42,21 @@ uint32_t Device::getSerialNumber() const {
+    canbus::Message message;
+    message.can_id = 0x600;
+    message.size = 8;
+    message.data[0] = 0x40;
```
master writes to slave

```suggestion
    message.data[0] = 0x23;
```

DeboraTahara committed:
fix: Set CD = 23 to write the id value
